### ISSUE TITLE
distutils dependency

### ISSUE THREAD

**Issue by faerytea** [McArcady/lnp-forge#99]
According to https://docs.python.org/3/whatsnew/3.12.html#whatsnew312-removed-distutils , `distutils` are removed from python 3.12.

`setuptools` are good drop-in replacement, so we can just replace dependency to `python3-distutils` to `python3-setuptools`. However, I only tested it on Debian trixie.

**Comment by ItsNilDev** [comment]
do you also get the 'nohup.out' error?

**Comment by faerytea** [comment]
Yes, but I cannot figure out what problems this might cause. Everything I use works like a clock.

**Comment by zahlman** [comment]
I had a look through the .deb to see how distutils is actually used in the current code. As far as I can tell, it's only in a single file:

core/df.py
```
# ...
from distutils import dir_util
# ...
def install_extras():
    # ...
    if not os.access(install_file, os.F_OK):
        log.i('Installing extras content for first time')
        dir_util.copy_tree(extras_dir, paths.get('df'))
        # ...
```

As noted, Distutils [was part of the standard library until Python 3.11](https://docs.python.org/3.11/distutils/apiref.html#distutils.dir_util.copy_tree), and removed in 3.12. Debian etc. packaged it separately, and now that package is gone matching the standard library removal.

The functionality described is just a utility for iterating over a directory and copying the files. It should be trivial to replace this with [shutil.copytree](https://docs.python.org/3.11/library/shutil.html#shutil.copytree) which has also been available in the standard library for approximately forever ([since at least 2.0](https://docs.python.org/2.0/lib/module-shutil.html)). So there's really no reason to have this dependency at all. Distutils (already extremely outdated at the time of removal) was intended to support Python tools used for packaging other Python code for distribution, so this use was also arguably a bit off-label.

However, I couldn't readily figure out which component provides this file...

**Comment by faerytea** [comment]
> However, I couldn't readily figure out which component provides this file...

[this](https://github.com/Pidgeot/python-lnp/blob/39d1db3acb3c663c56af39363df6654ab8a0535c/core/df.py#L120), from PyLNP (afaik, forge-lnp just [downloads release](https://github.com/McArcady/lnp-forge/blob/master/scripts/build/lnp.sh)). Should we patch it through regexp or make a PR to python-lnp?
